resources/models/reservation.py

```diff
@@ -357,6 +357,30 @@ def can_view_access_code(self, user):
             return True
         return self.resource.can_view_reservation_access_code(user)
 
+    def is_new_state_allowed(self, new_state, user):
+        """Checks that new state is permitted for this user."""
+
+        # if same state, always OK
+        if new_state == self.state:
+            return True
+
+        if not self.resource.can_approve_reservations(user):
```

Comment:
This is interesting. I thought (unpaid) reservations cancelled by the end user were cancelled with a `PUT` request too but apparently Varaamo sends a `DELETE` request instead.

Reply:
Yes was thinking whether `CANCELLED` needs to be here, as it's handled by another endpoint/set of permissions.

Reply:
Let's go with this. We might need to revisit this too when cancellations for paid reservations are implemented at some point.